theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+   * @param print_items if true include the list of items retained by the sketch
+   */
+  virtual string<A> to_string(bool print_items = false) const = 0;
+
+  /**
+   * This method serializes the sketch into a given stream in a binary form
+   * @param os output stream
+   */
+  virtual void serialize(std::ostream& os) const = 0;
+
+  // This is a convenience alias for users
+  // The type returned by the following serialize method
+  typedef vector_u8<A> vector_bytes;
+
+  /**
+   * This method serializes the sketch as a vector of bytes.
+   * An optional header can be reserved in front of the sketch.
+   * It is an uninitialized space of a given size.
+   * This header is used in Datasketches PostgreSQL extension.
+   * @param header_size_bytes space to reserve in front of the sketch
+   */
+  virtual vector_bytes serialize(unsigned header_size_bytes = 0) const = 0;
+
+  // This is a convenience alias for users
+  // The type returned by the following deserialize methods
+  // It is not possible to return instances of an abstract type, so this has to
+  // be a pointer
+  typedef std::unique_ptr<theta_sketch_dup_alloc<A>,
+                          std::function<void(theta_sketch_dup_alloc<A>*)>>
+      unique_ptr;
+
+  /**
+   * This method deserializes a sketch from a given stream.
+   * @param is input stream
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of a sketch as a unique_ptr
+   */
+  static unique_ptr deserialize(std::istream& is, uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * This method deserializes a sketch from a protobuf.
+   * @param pb input stream
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of a sketch as a unique_ptr
+   */
+  static unique_ptr deserialize(datasketches_pb::ThetaSketchDup& pb,
+                                uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * This method deserializes a sketch from a given array of bytes.
+   * @param bytes pointer to the array of bytes
+   * @param size the size of the array
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of the sketch
+   */
+  static unique_ptr deserialize(const void* bytes, size_t size,
+                                uint64_t seed = DEFAULT_SEED);
+
+  class const_iterator;
+
+  /**
+   * Iterator over hash values in this sketch.
+   * @return begin iterator
+   */
+  virtual const_iterator begin() const = 0;
+
+  /**
+   * Iterator pointing past the valid range.
+   * Not to be incremented or dereferenced.
+   * @return end iterator
+   */
+  virtual const_iterator end() const = 0;
+
+  /**
+   * @return true if *this equals r
+   */
+  virtual bool isEqual(const theta_sketch_dup_alloc& r) const;
+
+ protected:
+  enum flags { IS_BIG_ENDIAN, IS_READ_ONLY, IS_EMPTY, IS_COMPACT, IS_ORDERED };
+
+  bool is_empty_;
+  uint64_t theta_;
```

Comment:
could you add the comment regarding theta_?

Reply:
Updated. See [this commit](https://github.com/googleinterns/cardinality-estimation-with-data-deletions/pull/2/commits/d0bc1ab37365ff9b79bbcc8367ec7829c65e4035).